website/docs/guides/using-terraform-templates.mdx

```diff
@@ -92,3 +93,282 @@ gitops get template terraform tf-controller-aurora --list-parameters --endpoint
+          value: ${REGION}
+        - name: s3_import_bucket
+          value: ${S3_IMPORT_BUCKET}
+        varsFrom:
+        - kind: Secret
+          name: tf-controller-auth
+        sourceRef:
+          kind: GitRepository
+          name: flux-system
+          namespace: flux-system
+```
+
+Add the following Terraform manifest to the root of your Git repository.
+
+```yaml title="./rds.tf"
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+variable "cluster_identifier" {}
+variable "database_name" {}
+variable "master_username" {}
+variable "master_password" {}
+variable "backup_retention_period" {}
+variable "region" {}
+variable "s3_import_bucket" {}
+variable "aws_access_key" {}
+variable "aws_secret_key" {}
+variable "aws_role_arn" {}
+
+provider "aws" {
+  region = var.region
+  access_key = var.aws_access_key
+  secret_key = var.aws_secret_key
+
+  assume_role {
+    role_arn    = var.aws_role_arn
+  }
+}
+
+locals {
+  engine         = "aurora-mysql"
+  engine_version = "5.7.mysql_aurora.2.07.5"
+  port           = 3306
+}
+
+resource "aws_s3_bucket" "s3" {
+  bucket = var.s3_import_bucket
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_public_access_block" "s3" {
+  bucket = aws_s3_bucket.s3.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_object" "init_sql" {
+  bucket  = aws_s3_bucket.s3.id
+  key     = "init.sql.tar.gz"
+  content = <<EOM
+CREATE DATABASE test;
+CREATE TABLE test WITH COLUMNS id int, value string;
+EOM
+}
+
+resource "aws_rds_cluster" "mycluster" {
+  cluster_identifier      = var.cluster_identifier
+  engine                  = local.engine
+  engine_version          = local.engine_version
+  port                    = local.port
+  availability_zones      = ["us-west-2a", "us-west-2b", "us-west-2c"]
+  database_name           = var.database_name
+  master_username         = var.master_username
+  master_password         = var.master_password
+  backup_retention_period = var.backup_retention_period
+  skip_final_snapshot     = true
+  apply_immediately       = true
+
+  depends_on              = [aws_s3_bucket_object.init_sql]
+
+  s3_import {
+    source_engine         = "mysql"
+    source_engine_version = "5.7"
+    bucket_name           = aws_s3_bucket.s3.id
+    ingestion_role        = aws_iam_role.s3_rds.arn
+  }
+}
+
+resource "aws_rds_cluster_instance" "cluster_instance" {
+  count              = 1
+  identifier         = "${aws_rds_cluster.mycluster.id}-${count.index}"
+  cluster_identifier = aws_rds_cluster.mycluster.id
+  instance_class     = "db.t3.small"
+  engine             = aws_rds_cluster.mycluster.engine
+  engine_version     = aws_rds_cluster.mycluster.engine_version
+}
+
+resource "aws_iam_role" "s3_rds" {
+  name_prefix = "rds-s3-integration-role-"
+
+  assume_role_policy = <<EOF
```

Comment:
[not a blocker on merging]

AWS policy docs are best created using the [aws_iam_policy_document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) data source (they have some amount of in-built syntax checking and avoid having to deal with inline JSON), e.g.:

```hcl
resource "aws_iam_role" "s3_rds" {
  name_prefix        = "rds-s3-integration-role-"
  assume_role_policy = data.aws_iam_policy_document.assume_role.json
}

data "aws_iam_policy_document" "assume_role" {
  statement {
    effect = "Allow"
    actions = ["sts:assumeRole"]
    principals {
      type        = "Service"
      identifiers = ["rds.amazonaws.com"]
    }
  }
}
```